2000-11-07  Jakub Jelinek  <jakub@redhat.com>

	* typeck2.c (friendly_abort): Don't ICE if there were errors
	or not implemented features.

--- gcc/cp/typeck2.c.jj	Mon Jul 24 11:04:21 2000
+++ gcc/cp/typeck2.c	Tue Nov  7 17:50:00 2000
@@ -256,6 +256,12 @@ friendly_abort (where, file, line, func)
      int line;
      const char *func;
 {
+  if (errorcount > 0 || sorrycount > 0)
+    {
+      error ("confused by earlier errors, bailing out");
+      exit (34);
+    }
+
   if (where > 0)
     error ("Internal error #%d.", where);
 
